src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -1397,7 +1397,7 @@ public Object allocateUninitializedArray(Class<?> componentType, int length) {
            throw new IllegalArgumentException("Component type is not primitive");
        }
        if (length < 0) {
-           throw new IllegalArgumentException("Negative length");
+           throw new NegativeArraySizeException("Negative length" + length);
```

Comment:
This change is not relevant to this change, is it?

Reply:
This is to have consistent behaviour for all means of array declaration, otherwise e.g. `new byte[-1]` throws `NegativeArraySizeException` along with `Array.newInstance()`, but `Unsafe.allocateUninitializedArray()` throws `IllegalArgumentException` though it also allocates an array.

Reply:
Or maybe we should do it in a separate PR?